### DISCUSSION TITLE
OctoberCMS Support in docs, but no Driver included?

### DISCUSSION THREAD

**Discussion by philipbelcher** [beyondcode/herd-community#498] [Q&A]
Hi guys!

I can see October CMS [listed in the Herd Docs of supported CMSs](https://herd.laravel.com/docs/1/extending-herd/supported-frameworks#content-management-systems), but I was getting 404 errors on my October projects in Herd until I added a Custom OctoberCMS specific Driver to the /herd/config/drivers directory. (It's working _great_ now!)

Just thought I'd mention it in case it either needs removing from docs, the driver added to Herd or if I did something wrong!

Herd v1.5.0 (Build 23)

Many thanks!

**Comment by Johardmeier** [comment]
Herd Docs are not that good for windows. You should be able to create a LocalDriver in `C:\Users\YourName\.config\herd\config\valet\Drivers` but I kept running into missing files errors I couldn't resolve.
What worked is adding the missing OctoberCmsValetDriver.php directly into 
`C:\Program Files\Herd\resources\app.asar.unpacked\resources\valet\cli\Valet\Drivers\Specific`

```
  <?php
  namespace Valet\Drivers\Specific;  
  
  use Valet\Drivers\LaravelValetDriver;
  class OctoberCmsValetDriver extends LaravelValetDriver
  {
    public function serves($sitePath, $siteName, $uri):bool
    {
        if (file_exists($sitePath.'/modules/cms/routes.php')) {
            return true;
        }

        return false;
    }

    public function isStaticFile($sitePath, $siteName, $uri):? string
    {
        if (file_exists($staticFilePath = $sitePath.$uri)) {
            return $staticFilePath;
        }
        return false;
    }

    public function frontControllerPath(string $sitePath,string $siteName,string  $uri):? string
    {
        return $sitePath.'/index.php';
    }

  }

```

or you could add a file named `LocalValetDriver.php`  into the root directory of your october directory with the same content, but without the namespace (and obviously `class LocalValetDriver extends LaravelValetDriver`)
